src/lvl_script.c

```diff
@@ -1578,12 +1664,21 @@ void command_lose_game(void)
 void command_set_flag(long plr_range_id, const char *flgname, long val)
 {
     long flg_id = get_rid(flag_desc, flgname);
+    long flag_type = SVar_FLAG;
     if (flg_id == -1)
+    {
+        flg_id = get_rid(campaign_flag_desc, flgname);
+        if (flg_id != -1)
+        {
+            flag_type = SVar_CAMPAIGN_FLAG;
```

Comment:
@SimTheVoid This seems inconsistent. If I read this right, now the SET_FLAG supports FLAG and CAMPAIGN_FLAG, but not the other FLAG types you added that do work for ADD_TO_FLAG.